desktop/caliban.py

```diff
@@ -3887,18 +3886,19 @@ def label_mode_question_keypress_helper(self, symbol, modifiers):
 
         # RESPOND TO RELABEL QUESTION
         elif self.mode.action == "RELABEL":
-            if symbol == key.U:
-                self.action_relabel_unique()
-                self.mode.clear()
-            if symbol == key.P:
-                self.action_relabel_preserve()
-                self.mode.clear()
-            if symbol == key.S:
-                self.action_relabel_frame()
-                self.mode.clear()
             if symbol == key.SPACE:
                 self.action_relabel_all_frames()
                 self.mode.clear()
+            if not self.single_frame:
+                if symbol == key.U:
+                    self.action_relabel_unique()
+                    self.mode.clear()
+                if symbol == key.P:
+                    self.action_relabel_preserve()
+                    self.mode.clear()
+                if symbol == key.S:
+                    self.action_relabel_frame()
+                    self.mode.clear()
```

Comment:
These nested if blocks are very concerning. At some point, we're going to have to pay this technical debt. We are adding yet another conditional with this PR, and we need to call that out.

Reply:
Yeah, I don't love how keypresses are handled. This is definitely something I've been turning over in the back of my mind. I think this will be tied into or depend on a deeper refactor of `Mode` and may also involve creating specific event handlers https://pyglet.readthedocs.io/en/latest/programming_guide/events.html

Reply:
For this PR, though, I want to make sure there aren't hidden keypress options. Many of the single-frame actions wouldn't change the outcome of the action, but if the prompt indicates space to confirm/esc to cancel, I want space to be the only way to confirm. Functionally, I think the most important one of those is saving, as saving a single frame npz as a trk is totally useless.

Reply:
Can you please add this TODO to the Mode refactor issue, which should get bumped way up the priority list.